aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -1063,6 +1070,64 @@ std::tuple<Tensor, Tensor, Tensor, Tensor, c10::SymInt, c10::SymInt> _efficient_
     offset_t = at::empty({}, at::dtype(at::kLong).device(device));
   }
 
+#ifdef USE_ROCM
+  // ROCM Implementation
+  auto ret = aotriton::v2::flash::check_gpu(stream);
```

Comment:
do you need this if the check is already in sdp_utils?

Reply:
Oh I'm following the practice of `mha_fwd`, which also perform the checks for GPU architecture.
https://github.com/pytorch/pytorch/blob/dc8e3c2e904f22f05646e1d87987459af8845d7b/aten/src/ATen/native/transformers/cuda/flash_attn/flash_api.cpp#L359-L363

I'm not so sure what will happen if SDPA is call on GPUs with different arches (e.g., MI200+Navi31). This check is the last line of defense to guarantee the target GPU does support the SDPA request.